lib/elixir/lib/kernel/typespec.ex

```diff
@@ -883,7 +887,9 @@ defmodule Kernel.Typespec do
 
   # Handle remote calls
   defp typespec({{:., meta, [remote, name]}, _, args} = orig, vars, caller) do
-    remote = Macro.expand remote, caller
+    # We set a function name to avoid tracking
+    # aliases in typespecs as compile time dependencies.
+    remote = Macro.expand remote, %{caller | function: {:typespec, 0}}
```

Comment:
I thought the fix would be something exactly like that. :D

Should we do `__info__` or `__typespec__` as the function name though? Another option is to disable the lexical tracker since it is a field in the environment and that would disable the tracking.